packages/drivers/odsp-driver/src/odspUtils.ts

```diff
@@ -195,12 +192,16 @@ export async function fetchAndParseAsJSONHelper<T>(
         };
         return res;
     } catch (e) {
+        // JSON.parse() can fail and message would container full request URI, including
+        // tokens... It fails for me with "Unexpected end of JSON input" quite often - an attempt to download big file
+        // (many ops) almost always ends up with this error - I'd guess 1% of op request end up here... It always
+        // succeeds on retry.
+        // So do not log error object itself.
         throwOdspNetworkError(
             "errorWhileParsingFetchResponse",
             fetchIncorrectResponse,
```

Comment:
I think you can get rid of this if you move try-catch to only be around JSON.parse.  Then you can read the status code off the response and log it.

Reply:
Get rid of what? I'll reduce the scope of try/catch, but I'd prefer to keep the comment around

Reply:
Get rid of `fetchIncorrectResponse`

Reply:
I thought about it but stopped short of making it happen. That's the only place where we have fake code but real response. So I did not feel comfortable understanding and ensuring we do not lose anything by not going through  throwOdspNetworkError. I think we need to do it, but there are questions to be answered like - can it happen that SPO returns 200 but it's an error and response contains facetCodes - wouldn't we want to record it in telemetry to be able to better analyze it?

Feel free to open an issue, but I feel like it's rather low gain at this point - all the key usage is much better now!